### ISSUE TITLE
Images disappearing from the output?

### ISSUE THREAD

**Issue by sdlovecraft** [phillipi/pix2pix#76]
I'm attempting to process large amounts of images  at a time (7000+) with pix2pix, but I am finding that Images from the /val folder are getting lost by the time they are processed into the `results/../../input folder`.  If i take the same images that were lost, and put them back into pix2pix, they are processed correctly - so it isn't a problem with file corruption. 

Any ideas?

**Comment by tyshiwo** [comment]
I met the same problem when processing large amounts of images! I found it loaded some images twice, therefor some other images are skipped!

**Comment by sdlovecraft** [comment]
Yeah I have just written a script that checks the differences between the input folder and the output folder and runs the test process again with missing files, but definitely a bug!

**Comment by phillipi** [comment]
Thanks for finding this bug! I'll look into it when I get a chance. @sdlovecraft do you mind sharing your script in the meantime?

**Comment by sdlovecraft** [comment]
@phillipi  yeah of course!

So this script assumes that you have moved the old val files into a folder called 'val_old' and all your output files into a folder called 'output_old', it then looks at them and compares the values to the output, and places any missing files into a newly created val folder. You then just have to run the test process again. 

```
 import os
  
  BASE_DIR = '~/Code/pix2pix'
  PIX_2_PIX_OUTPUT_DIR = BASE_DIR+'/results/result_name/latest_net_G_val/images/output_old'
  INPUT_DIR = BASE_DIR+'/data/val_old'
  VAL_DIR = BASE_DIR+'/data/val'
  
  if not os.path.exists(VAL_DIR):
      os.mkdir(VAL_DIR)
  
  
  def diff(first, second):
      second = set(second)
      return [INPUT_DIR+item for item in first
              if item.split('/')[-1] not in second]
  
  
  def get_filenames(dir_name):
      return [fname
              for fname in os.listdir(dir_name)
              if fname.endswith('.png')]
  
  
  input_filenames = get_filenames(INPUT_DIR)
  p2p_filenames = get_filenames(PIX_2_PIX_OUTPUT_DIR)
  
  missing_filenames = diff(input_filenames, p2p_filenames)
  
  for filename in missing_filenames:
      new_name = VAL_DIR + filename.split('/')[-1]
      print('moving from', filename, 'moving to', new_name )
      os.rename(filename, new_name)
                                  
```

**Comment by phillipi** [comment]
thanks!

**Comment by uakfdotb** [comment]
I encountered this bug a few days ago too, I think #79 will fix it.

**Comment by janyasir** [comment]
I have tried #79 but it is not working for me. 
I have 20460 test images and only 19185 images are generated at the output. Some random files are skipped. It worked perfectly for less size of datasets. Plz guide me. Thanks.

**Comment by uakfdotb** [comment]
@janyasir if you find this line:

    self.classList[i] = torch.linspace(runningIndex + 1, runningIndex + length, length):long()

And then add after it:

    print(self.classList[i])

In the output where it prints that, does it have some skipped numbers (e.g. repeats 2000 twice and then skips 2001)?

Also for now you could use the code that @sdlovecraft posted to rerun it on the missing files.

**Comment by janyasir** [comment]
@uakfdotb I tried the command print(self.classList[i]). Because of the screen scrolling limit, I could see last values ranging from 19476:20460. There is no repitiiton, and they are in a sequence. And it also prints in the end [torch.LongTensor of size 20460]. It seems like it does pick up all 20460 files, but tests on only 19185 files generated. Or maybe testing them all but not writing all the results back to the folder.

I will try the other code for a while. Thankyou for the reply.

**Comment by samhains** [comment]
#79 seemed to fix it for me.. thanks!!!

**Comment by janyasir** [comment]
#79 working for me too... I was doing a mistake
Incorrect way: self.classList[i] = torch.range(runningIndex + 1, runningIndex + length, length):long()
Correct way : self.classList[i] = torch.range(runningIndex + 1, runningIndex + length):long()
